### DISCUSSION TITLE
How to distinguish TE and TM

### DISCUSSION THREAD

**Discussion by niubiworker** [NanoComp/meep#2703] [Q&A]
I did not find any discussion or guidance on how to distinguish `TE` and `TM` in the relevant documents, but I believe this is very important. Can anyone help me distinguish `TE` and `TM` of light sources in simulation

**Comment by stevengj** [comment]
This is simply set by the polarization of the current source.  (There are also options for eigenmode sources to set the symmetry (hence polarization) of the modes.)

Just beware that what people mean by "TE" and "TM" polarizations vary in the literature.  In 2d, I tend to use "Ez polarization" and "Hz polarization" these days to be less ambiguous.  People also talk about "s" and "p" polarization for planewaves with electric field perpendicular to or lying within the plane of incidence, respectively (corresponding to Ez and Hz in 2d).
